### PR TITLE
Don't use a pipe for output capturing in Python's test runner

### PR DESCRIPTION
**Pull request by soltanmm**
Leci n'est pas une pipe

### CODE REVIEW

**Comment by soltanmm** [comment]
May or may not fix #5779, we'll see.

**Comment by soltanmm** [comment]
Python green.

File: src/python/grpcio/tests/_runner.py [hunk 1]
```diff
@@ -43,13 +44,6 @@
...
 
-# This number needs to be large enough to outpace output on stdout and stderr
-# from the gRPC core, otherwise we could end up in a potential deadlock. This
-# stems from the OS waiting on someone to clear a filled pipe buffer while the
-# GIL is held from a write to stderr from gRPC core, but said someone is in
-# Python code thus necessitating GIL acquisition.
-_READ_BYTES = 2**20
-
 
 class CapturePipe(object):
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/tests/_runner.py]: Any need to rename this class?

File: src/python/grpcio/tests/_runner.py [hunk 2]
```diff
@@ -43,13 +44,6 @@
...
   """A context-manager pipe to redirect output to a byte array.
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/tests/_runner.py]: How about removing no-longer-existent class-private attributes from this class-public doc string?

Any need to update the summary line?

**Comment by nathanielmanistaatgoogle** [comment]
Optional: consider converting `output` from a public field to an internal field with a public accessor method that raises an exception if called at the wrong stage of the CapturePipe object life cycle.

**Comment by soltanmm** [comment]
Turns out that I'd forgotten that the 'output' basically needs to be available at all times (e.g. if the user interrupts the test runner before anything's happened).

**Comment by soltanmm** [comment]
Python

**Comment by jtattermusch** [comment]
Both failures are known flakes.

**Pull request merged**